pallets/vesting/src/lib.rs

```diff
@@ -0,0 +1,271 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+use core::convert::TryFrom;
+use frame_support::{
+	ensure,
+	pallet_prelude::*,
+	traits::{
+		Currency, ExistenceRequirement, Get, LockIdentifier, LockableCurrency, UnixTime,
+		WithdrawReasons,
+	},
+};
+use frame_system::{ensure_signed, pallet_prelude::*};
+pub use pallet::*;
+use sp_runtime::{
+	traits::{Saturating, StaticLookup, Zero},
+	Percent,
+};
+
+type BalanceOf<T> =
+	<<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+pub type Schedule = u64;
+
+const VESTING_ID: LockIdentifier = *b"mantavst";
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// The overarching event type.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		/// The currency trait.
+		type Currency: LockableCurrency<Self::AccountId>;
+
+		type Timestamp: UnixTime;
+
+		/// The minimum amount transferred to call `vested_transfer`.
+		#[pallet::constant]
+		type MinVestedTransfer: Get<BalanceOf<Self>>;
+
+		/// The maximum length of schedule is allowed.
+		#[pallet::constant]
+		type MaxScheduleLength: Get<u32>;
+	}
+
+	/// Information regarding the vesting of a given account.
+	#[pallet::storage]
+	#[pallet::getter(fn vesting_balance)]
+	pub(super) type VestingBalances<T: Config> =
+		StorageMap<_, Blake2_128Concat, T::AccountId, BalanceOf<T>>;
+
+	/// Information regarding the vesting of a given account.
+	#[pallet::storage]
+	#[pallet::getter(fn vesting_schedule)]
+	pub(super) type VestingSchedule<T: Config> = StorageValue<
+		_,
+		BoundedVec<(Percent, Schedule), T::MaxScheduleLength>,
```

Comment:
Why not just use `T:Timestamp` here? (Instead of `Schedule`, which is `u64`)

Reply:
It could not be compiled with this definition:
```rust
type Timestamp: Parameter + UnixTime + From<u64> + Into<u64>;
```
Because `pallet-timestamp` doesn't implement `From/Into` trait. Lots of compilation errors will be prompted.